src/daemon/stripeDaemon.ts

```diff
@@ -0,0 +1,122 @@
+import * as grpc from '@grpc/grpc-js';
+import * as vscode from 'vscode';
+import {DaemonConfig, MalformedConfigError, NoDaemonCommandError, isDaemonConfig} from './types';
+import {authOutboundInterceptor, errorInboundInterceptor} from './interceptors';
+import {LineStream} from 'byline';
+import {StripeCLIClient} from '../rpc/commands_grpc_pb';
+import {StripeClient} from '../stripeClient';
+import {Writable} from 'stream';
+import execa from 'execa';
+
+/**
+ * StripeDaemon handles starting and restarting the Stripe daemon gRPC server and creating the
+ * client connection.
+ */
+export class StripeDaemon {
+  private stripeClient: StripeClient;
```

Comment:
We only need this for the getCLIPath currently right?  In the future, will this daemon replace `StripeClient` entirely?

Reply:
Yep, we only need `getCLIPath` currently. The `StripeClient` definitely has tech debt and I could see `StripeDaemon` replacing it. I could also see `StripeClient` still existing, but with only `getCLIPath`, just to minimize concerns. I guess I'm ok with either but I might prefer the latter, I'm not sure.

Reply:
Actually sorry, it's not the daemon that should replace the client, since the daemon's purpose is to manage that process -- should StripeClient actually be depending on the daemon? What would be the entry-point class that manages the requests to "stripe" -- that class should make it so that the clients (our other vscode classes) are unaware of whether we are communicating via stripe by manually kicking off cli commands via the terminal or via gRPC.

This is all non-blocking I am just thinking outloud